### PR TITLE
Ryna lb111 ag resbio r c

### PR DESCRIPTION
**Pull request by rynacui**

### CODE REVIEW

File: R/zchunk_LB111.ag_resbio_R_C.R [hunk 1]
```diff
@@ -8,13 +8,14 @@
 #' a vector of output names, or (if \code{command} is "MAKE") all
```
Comment (bpbond) [inline R/zchunk_LB111.ag_resbio_R_C.R]: Don't forget to fill out line 3 please!

File: R/zchunk_LB111.ag_resbio_R_C.R [hunk 2]
```diff
@@ -32,39 +33,47 @@ module_aglu_LB111.ag_resbio_R_C_DISABLED <- function(command, ...) {
...
+      filter(year %in% FINAL_HISTORICAL_YEAR) %>%
+      select(-year) %>%
+      rename(prod = value) %>%
+      full_join(Various_ag_resbio_data, by = "item") %>%
+    # Drop rows with NA values ( dropping commodities that are not in the resbio dataset )
+      na.omit() %>%
+    # also drop rows where production weights are zero, as these would return missing values later on
+      filter(prod != 0) %>%
+    # Multiply by production to get weights, change to long-format for easier calculation
+      gather(resbio_params, value, -c(iso, item, prod)) %>%
```
Comment (bpbond) [inline R/zchunk_LB111.ag_resbio_R_C.R]: `gather(resbio_params, value, -iso, -item, -prod)` probably clearer?

File: R/zchunk_LB111.ag_resbio_R_C.R [hunk 3]
```diff
@@ -32,39 +33,47 @@ module_aglu_LB111.ag_resbio_R_C_DISABLED <- function(command, ...) {
...
+      mutate(value = value * prod) %>%
+    # Add vectors for GCAM regions and commodities, collapse, and divide by production to get residue biomass values
+      left_join(iso_GCAM_regID[c( "iso", "GCAM_region_ID")], by = "iso") %>%
```
Comment (bpbond) [inline R/zchunk_LB111.ag_resbio_R_C.R]: This should probably be a `left_join_error_no_match`, which throws an error if there's not a 1-to-1 match for everything.

File: R/zchunk_LB111.ag_resbio_R_C.R [hunk 4]
```diff
@@ -32,39 +33,47 @@ module_aglu_LB111.ag_resbio_R_C_DISABLED <- function(command, ...) {
...
+      left_join(FAO_ag_items_PRODSTAT[c( "item", "GCAM_commodity")], by = "item") %>%
```
Comment (bpbond) [inline R/zchunk_LB111.ag_resbio_R_C.R]: Probably also `left_join_error_no_match`? Every item has to have a `GCAM_commodity` entry right?

File: R/zchunk_LB111.ag_resbio_R_C.R [hunk 5]
```diff
@@ -32,39 +33,47 @@ module_aglu_LB111.ag_resbio_R_C_DISABLED <- function(command, ...) {
...
+      mutate(value = value / prod) %>%
+      select(-prod) %>%
+      spread(resbio_params, value) %>%
 
     # Produce outputs
     # Temporary code below sends back empty data frames marked "don't test"
     # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
     # There's also a `same_precursors_as(x)` you can use
     # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
```
Comment (bpbond) [inline R/zchunk_LB111.ag_resbio_R_C.R]: Please remove commented lines above, except for the first.

File: R/zchunk_LB111.ag_resbio_R_C.R [hunk 6]
```diff
@@ -32,39 +33,47 @@ module_aglu_LB111.ag_resbio_R_C_DISABLED <- function(command, ...) {
...
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+      add_title("Weighted average residue biomass parameters by GCAM region / commodity") %>%
+      add_units("units: varied") %>%
+      add_comments("Weighted average residue biomass parameters by GCAM region / commodity") %>%
+      add_comments("Units: varied") %>%
```
Comment (bpbond) [inline R/zchunk_LB111.ag_resbio_R_C.R]: Don't need this.

File: R/zchunk_LB111.ag_resbio_R_C.R [hunk 7]
```diff
@@ -32,39 +33,47 @@ module_aglu_LB111.ag_resbio_R_C_DISABLED <- function(command, ...) {
...
     # There's also a `same_precursors_as(x)` you can use
     # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+      add_title("Weighted average residue biomass parameters by GCAM region / commodity") %>%
+      add_units("units: varied") %>%
+      add_comments("Weighted average residue biomass parameters by GCAM region / commodity") %>%
```
Comment (bpbond) [inline R/zchunk_LB111.ag_resbio_R_C.R]: Probably change this to a 1-2 sentence description of the calculation process, not the output.

File: R/constants.R [hunk 8]
```diff
@@ -17,6 +17,7 @@ FLAG_NO_TEST    <- "NO_TEST"
 # Time constants
 HISTORICAL_YEARS <- 1971:2010
 FUTURE_YEARS <- seq( 2015, 2100, 5 )
+FINAL_HISTORICAL_YEAR <- 2010
```
Comment (kvcalvin) [inline R/constants.R]: Should we calculate this from `HISTORICAL_YEARS` so we don't have to update both fields if the years change?

File: inst/extdata/aglu/Various_ag_resbio_data.csv [hunk 9]
```diff
@@ -1,7 +1,8 @@
 # File: Various_ag_resbio_data.csv,,,,,
+# Title: Various_ag_resbio_data,,,,,
+# Units: Varied,,,,,
```
Comment (kvcalvin) [inline inst/extdata/aglu/Various_ag_resbio_data.csv]: Could you fill in the units for the various columns?

File: R/zchunk_LB111.ag_resbio_R_C.R (hunk 7 above)
Comment (kvcalvin) [inline R/zchunk_LB111.ag_resbio_R_C.R]: In addition, can you clarify in the comments what the residue biomass parameters are?

File: R/constants.R (hunk 8 above)
Comment (bpbond) [inline R/constants.R]: Oh, definitely. Good catch.

File: inst/extdata/aglu/Various_ag_resbio_data.csv (hunk 9 above)
Comment (rynacui) [inline inst/extdata/aglu/Various_ag_resbio_data.csv]: Not all units are specified in the input file or the original code. The five parameters are: HarvestIndex, ErosCtrl_tHa, ResEnergy_GJt, Root_Shoot, and WaterContent. So I guess ErosCtrl is in ton/Ha and ResEnergy is in GJ/ton? But I'm not sure the other three. Probably have to go to the aglu data documentation.

File: R/constants.R (hunk 8 above)
Comment (rynacui) [inline R/constants.R]: Great idea! Should I use `max(HISTORICAL_YEARS)`?

**Comment by bpbond** [comment]
@kvcalvin look good?

**Pull request merged**